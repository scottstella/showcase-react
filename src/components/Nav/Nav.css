.nav {
  background-color: #b6c199;
  padding: 30px;
  /*min-height: 100%; this causes aside to bleed into nave at 1200px*/
  max-width: 100%;
  display: flex;
  flex-direction: column;
  text-decoration: underline;
}
