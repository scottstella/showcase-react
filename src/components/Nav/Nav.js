import React from "react";
import "./Nav.css";

export default function Nav() {
  return (
    <div className="nav">
      <p>Admin</p>
      <li>Manage Meta-Data</li>
      <li>Manage Cards</li>

      <p>Decks</p>
      <li>Maintain</li>

      <p>Admin</p>
      <li>Manage Meta-Data</li>
      <li>Manage Cards</li>
    </div>
  );
}
