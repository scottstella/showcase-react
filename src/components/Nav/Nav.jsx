import React, { useState } from "react";
import "./Nav.css";
import { NavLink } from "react-router-dom";

export default function Nav() {
  const [isAdminCollapsed, setIsAdminCollapsed] = useState(true);
  const [isDecksCollapsed, setIsDecksCollapsed] = useState(true);

  const clickAdmin = () => {
    setIsAdminCollapsed(!isAdminCollapsed);
  };
  const clickDecks = () => {
    setIsDecksCollapsed(!isDecksCollapsed);
  };

  return (
    <div className="nav">
      <h2>Menu</h2>
      <div className="nav-collapsable">
        <i
          className={`${
            isAdminCollapsed
              ? "fa-solid fa-angle-right"
              : "fa-solid fa-angle-down"
          }`}
        />
        <div onClick={clickAdmin}>Admin</div>
      </div>
      <div
        className={`${
          isAdminCollapsed ? "nav-group-collapsed" : "nav-group-expanded"
        }`}
      >
        <NavLink to="/ManageMetaData" className="nav-link">
          Manage Meta-Data
        </NavLink>

        <NavLink to="/ManageMetaData" className="nav-link">
          Manage Cards
        </NavLink>
      </div>
      <div className="nav-collapsable">
        <i
          className={`${
            isDecksCollapsed
              ? "fa-solid fa-angle-right"
              : "fa-solid fa-angle-down"
          }`}
        />
        <div onClick={clickDecks}>Decks</div>
      </div>
      <div
        className={`${
          isDecksCollapsed ? "nav-group-collapsed" : "nav-group-expanded"
        }`}
      >
        <NavLink to="/ManageMetaData" className="nav-link">
          Link 1
        </NavLink>

        <NavLink to="/ManageMetaData" className="nav-link">
          Link 2
        </NavLink>
      </div>
    </div>
  );
}
