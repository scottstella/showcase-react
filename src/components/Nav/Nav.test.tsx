import React from "react";
import { describe, it, expect } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Nav from "./Nav";

// Wrap component with Router for NavLink to work
const renderWithRouter = () => {
  return render(
    <BrowserRouter>
      <Nav />
    </BrowserRouter>
  );
};

describe("Nav", () => {
  it("renders without crashing", () => {
    renderWithRouter();
    expect(screen.getByText("Menu")).toBeInTheDocument();
  });

  it("renders all navigation sections", () => {
    renderWithRouter();
    expect(screen.getByText("Admin")).toBeInTheDocument();
    expect(screen.getByText("Decks")).toBeInTheDocument();
  });

  describe("Admin section", () => {
    it("starts collapsed", () => {
      renderWithRouter();
      const adminLinks = screen.getByText("Manage Meta-Data").parentElement;
      expect(adminLinks).toHaveClass("nav-group-collapsed");
      expect(screen.getByText("Admin").previousElementSibling).toHaveClass(
        "fa-angle-right"
      );
    });

    it("expands when clicked", () => {
      renderWithRouter();
      fireEvent.click(screen.getByText("Admin"));
      const adminLinks = screen.getByText("Manage Meta-Data").parentElement;
      expect(adminLinks).toHaveClass("nav-group-expanded");
      expect(screen.getByText("Admin").previousElementSibling).toHaveClass(
        "fa-angle-down"
      );
    });

    it("shows correct links when expanded", () => {
      renderWithRouter();
      fireEvent.click(screen.getByText("Admin"));

      const links = screen.getAllByText(/Manage/);
      expect(links).toHaveLength(2);
      expect(links[0]).toHaveTextContent("Manage Meta-Data");
      expect(links[1]).toHaveTextContent("Manage Cards");
    });
  });

  describe("Decks section", () => {
    it("starts collapsed", () => {
      renderWithRouter();
      const decksLinks = screen.getByText("Link 1").parentElement;
      expect(decksLinks).toHaveClass("nav-group-collapsed");
      expect(screen.getByText("Decks").previousElementSibling).toHaveClass(
        "fa-angle-right"
      );
    });

    it("expands when clicked", () => {
      renderWithRouter();
      fireEvent.click(screen.getByText("Decks"));
      const decksLinks = screen.getByText("Link 1").parentElement;
      expect(decksLinks).toHaveClass("nav-group-expanded");
      expect(screen.getByText("Decks").previousElementSibling).toHaveClass(
        "fa-angle-down"
      );
    });

    it("shows correct links when expanded", () => {
      renderWithRouter();
      fireEvent.click(screen.getByText("Decks"));

      const links = screen.getAllByText(/Link/);
      expect(links).toHaveLength(2);
      expect(links[0]).toHaveTextContent("Link 1");
      expect(links[1]).toHaveTextContent("Link 2");
    });
  });

  it("allows sections to be independently expanded", () => {
    renderWithRouter();

    // Expand both sections
    fireEvent.click(screen.getByText("Admin"));
    fireEvent.click(screen.getByText("Decks"));

    // Check both are expanded
    expect(screen.getByText("Manage Meta-Data").parentElement).toHaveClass(
      "nav-group-expanded"
    );
    expect(screen.getByText("Link 1").parentElement).toHaveClass(
      "nav-group-expanded"
    );

    // Collapse Admin only
    fireEvent.click(screen.getByText("Admin"));

    // Check Admin is collapsed but Decks is still expanded
    expect(screen.getByText("Manage Meta-Data").parentElement).toHaveClass(
      "nav-group-collapsed"
    );
    expect(screen.getByText("Link 1").parentElement).toHaveClass(
      "nav-group-expanded"
    );
  });

  it("matches snapshot", () => {
    const { container } = renderWithRouter();
    expect(container).toMatchSnapshot();
  });
});
