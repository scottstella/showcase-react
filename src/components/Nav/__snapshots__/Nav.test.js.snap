// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Nav > matches snapshot 1`] = `
<div>
  <div
    class="nav"
  >
    <h2>
      Menu
    </h2>
    <div
      class="nav-collapsable"
    >
      <i
        class="fa-solid fa-angle-right"
      />
      <div>
        Admin
      </div>
    </div>
    <div
      class="nav-group-collapsed"
    >
      <a
        class="nav-link"
        data-discover="true"
        href="/ManageMetaData"
      >
        Manage Meta-Data
      </a>
      <a
        class="nav-link"
        data-discover="true"
        href="/ManageMetaData"
      >
        Manage Cards
      </a>
    </div>
    <div
      class="nav-collapsable"
    >
      <i
        class="fa-solid fa-angle-right"
      />
      <div>
        Decks
      </div>
    </div>
    <div
      class="nav-group-collapsed"
    >
      <a
        class="nav-link"
        data-discover="true"
        href="/ManageMetaData"
      >
        Link 1
      </a>
      <a
        class="nav-link"
        data-discover="true"
        href="/ManageMetaData"
      >
        Link 2
      </a>
    </div>
  </div>
</div>
`;
